python/ray/experimental/workflow/step_executor.py

```diff
@@ -145,15 +132,116 @@ def commit_step(store: workflow_storage.WorkflowStorage,
     store.save_step_output(step_id, ret, outer_most_step_id)
 
 
-@ray.remote
+class StepType(enum.Enum):
+    """All step types."""
+    FUNCTION = 1
+    ACTOR_METHOD = 2
+    READONLY_ACTOR_METHOD = 3
+
+
+class ExecutionResult:
```

Comment:
Let's call it `ExecutionUnit`. The reason is that it's not just about the result, it contains two parts: data + logic to execute.